File: number-of-burgers-with-no-waste-of-ingredients/number-of-burgers-with-no-waste-of-ingredients.cpp
class Solution {
public:
    vector<int> numOfBurgers(int tomatoSlices, int cheeseSlices) {
        if(tomatoSlices < 2*cheeseSlices || (tomatoSlices - 2*cheeseSlices)%2 != 0){
            return {};
        }else{
            if(4*cheeseSlices < tomatoSlices || (4*cheeseSlices - tomatoSlices)%2)return {};
            return {(tomatoSlices - 2*cheeseSlices)/2, (4*cheeseSlices - tomatoSlices)/2};
        }
    }
};

File: number-of-burgers-with-no-waste-of-ingredients/number-of-burgers-with-no-waste-of-ingredients.go
func numOfBurgers(tomatoSlices int, cheeseSlices int) []int {
    if(tomatoSlices < 2*cheeseSlices || (tomatoSlices - 2*cheeseSlices)%2 != 0){
        return []int{}
    }else{
        if(4*cheeseSlices < tomatoSlices || (4*cheeseSlices - tomatoSlices)%2 != 0) {
            return []int{}
        }
        
        return []int{(tomatoSlices - 2*cheeseSlices)/2, (4*cheeseSlices - tomatoSlices)/2}
    }
}